### ISSUE TITLE
Deadlock loading archive server

### ISSUE THREAD

**Issue by eclipse-sailing-analytics-bot** [eclipse-sailing-analytics/sailing-analytics#4414]
| Field | Value |
|---|---|
| **Reporter** | @axeluhl |
| **Assignee** | @axeluhl |
| **Product** | Sailing Race Analytics |
| **Component** | Server |
| **Version** | unspecified |
| **Hardware** | All |
| **OS** | All |
| **Severity** | normal |
| **Priority** | P1 |
| **CC (unmapped)** | Steffen Schäfer (`steffen.schaefer@pgt.de`) |
| **Created** | 2018-01-13T15:51:49Z |
| **Original Bug** | [Bug 4414](https://old-bugzilla.sapsailing.com/bugzilla/show_bug.cgi?id=4414) |

---

Jan 13, 2018 3:49:10 PM com.sap.sse.concurrent.LockUtil lock
INFO: Couldn't acquire lock readLock trackeRaces lock for tracked regatta SCL 2017 - Porto Cervo (3) in 5s in thread com.sap.sailing.domain.tractracadapter.impl.RawPositionReceiver at java.lang.Thread.getStackTrace(Thread.java:1628)
com.sap.sse.concurrent.LockUtil.getCurrentStackTrace(LockUtil.java:371)
com.sap.sse.concurrent.LockUtil.lock(LockUtil.java:401)
com.sap.sse.concurrent.LockUtil.acquireLockVirtuallyOrActually(LockUtil.java:114)
com.sap.sse.concurrent.LockUtil.lockForRead(LockUtil.java:92)
com.sap.sailing.domain.tracking.impl.TrackedRegattaImpl.lockTrackedRacesForRead(TrackedRegattaImpl.java:68)
com.sap.sailing.domain.tracking.impl.TrackedRegattaImpl.getExistingTrackedRace(TrackedRegattaImpl.java:198)
com.sap.sailing.domain.tracking.impl.DynamicTrackedRegattaImpl.getExistingTrackedRace(DynamicTrackedRegattaImpl.java:32)
com.sap.sailing.domain.tracking.impl.DynamicTrackedRegattaImpl.getExistingTrackedRace(DynamicTrackedRegattaImpl.java:1)
com.sap.sailing.domain.tracking.impl.TrackedRegattaImpl.getTrackedRace(TrackedRegattaImpl.java:162)
com.sap.sailing.domain.tracking.impl.DynamicTrackedRegattaImpl.getTrackedRace(DynamicTrackedRegattaImpl.java:21)
com.sap.sailing.domain.tractracadapter.impl.AbstractReceiverWithQueue.getTrackedRace(AbstractReceiverWithQueue.java:237)
com.sap.sailing.domain.tractracadapter.impl.RawPositionReceiver.handleEvent(RawPositionReceiver.java:59)
com.sap.sailing.domain.tractracadapter.impl.AbstractReceiverWithQueue.run(AbstractReceiverWithQueue.java:172)
java.lang.Thread.run(Thread.java:808)

The current writer is:
Thread[Default background executor pool-2-thread-9,4,main]
com.sap.sailing.domain.tracking.AbstractTrackedRegattaAndRaceObserver$RegattaListener.raceRemoved(AbstractTrackedRegattaAndRaceObserver.java:101)
com.sap.sailing.domain.tracking.AbstractTrackedRegattaAndRaceObserver$RegattaListener$1.raceRemoved(AbstractTrackedRegattaAndRaceObserver.java:89)
com.sap.sailing.domain.tracking.impl.TrackedRegattaImpl.removeTrackedRace(TrackedRegattaImpl.java:139)
com.sap.sailing.server.impl.RacingEventServiceImpl.removeRace(RacingEventServiceImpl.java:2439)
com.sap.sailing.domain.tractracadapter.impl.TracTracRaceTrackerImpl.<init>(TracTracRaceTrackerImpl.java:398)
com.sap.sailing.domain.tractracadapter.impl.TracTracRaceTrackerImpl.<init>(TracTracRaceTrackerImpl.java:267)
com.sap.sailing.domain.tractracadapter.impl.DomainFactoryImpl.createRaceTracker(DomainFactoryImpl.java:710)
com.sap.sailing.domain.tractracadapter.impl.RaceTrackingConnectivityParametersImpl.createRaceTracker(RaceTrackingConnectivityParametersImpl.java:120)
com.sap.sailing.server.impl.RacingEventServiceImpl.addRace(RacingEventServiceImpl.java:1571)
com.sap.sailing.server.impl.RacingEventServiceImpl.lambda$4(RacingEventServiceImpl.java:791)
com.sap.sailing.server.impl.RacingEventServiceImpl$$Lambda$30/450464248.accept(Unknown Source)
com.sap.sailing.domain.persistence.impl.DomainObjectFactoryImpl.lambda$2(DomainObjectFactoryImpl.java:2381)
com.sap.sailing.domain.persistence.impl.DomainObjectFactoryImpl$$Lambda$39/2053027844.withService(Unknown Source)
com.sap.sse.osgi.CachedOsgiTypeBasedServiceFinder.applyServiceWhenAvailable(CachedOsgiTypeBasedServiceFinder.java:82)
com.sap.sailing.domain.persistence.impl.DomainObjectFactoryImpl.lambda$1(DomainObjectFactoryImpl.java:2370)
com.sap.sailing.domain.persistence.impl.DomainObjectFactoryImpl$$Lambda$38/270305695.run(Unknown Source)
java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
java.util.concurrent.FutureTask.run(FutureTask.java:266)
java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
java.util.concurrent.FutureTask.run(FutureTask.java:266)
java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
com.sap.sse.util.impl.ThreadPoolAwareRunnableScheduledFutureDelegate.run(ThreadPoolAwareRunnableScheduledFutureDelegate.java:23)
java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
java.lang.Thread.run(Thread.java:808)

Trying again...
Jan 13, 2018 3:49:10 PM com.sap.sse.concurrent.LockUtil lock
INFO: Couldn't acquire lock readLock trackeRaces lock for tracked regatta SCL 2017 - Porto Cervo (3) in 5s in thread com.sap.sailing.domain.tractracadapter.impl.RawPositionReceiver at java.lang.Thread.getStackTrace(Thread.java:1628)
com.sap.sse.concurrent.LockUtil.getCurrentStackTrace(LockUtil.java:371)
com.sap.sse.concurrent.LockUtil.lock(LockUtil.java:401)
com.sap.sse.concurrent.LockUtil.acquireLockVirtuallyOrActually(LockUtil.java:114)
com.sap.sse.concurrent.LockUtil.lockForRead(LockUtil.java:92)
com.sap.sailing.domain.tracking.impl.TrackedRegattaImpl.lockTrackedRacesForRead(TrackedRegattaImpl.java:68)
com.sap.sailing.domain.tracking.impl.TrackedRegattaImpl.getExistingTrackedRace(TrackedRegattaImpl.java:198)
com.sap.sailing.domain.tracking.impl.DynamicTrackedRegattaImpl.getExistingTrackedRace(DynamicTrackedRegattaImpl.java:32)
com.sap.sailing.domain.tracking.impl.DynamicTrackedRegattaImpl.getExistingTrackedRace(DynamicTrackedRegattaImpl.java:1)
com.sap.sailing.domain.tracking.impl.TrackedRegattaImpl.getTrackedRace(TrackedRegattaImpl.java:162)
com.sap.sailing.domain.tracking.impl.DynamicTrackedRegattaImpl.getTrackedRace(DynamicTrackedRegattaImpl.java:21)
com.sap.sailing.domain.tractracadapter.impl.AbstractReceiverWithQueue.getTrackedRace(AbstractReceiverWithQueue.java:237)
com.sap.sailing.domain.tractracadapter.impl.RawPositionReceiver.handleEvent(RawPositionReceiver.java:59)
com.sap.sailing.domain.tractracadapter.impl.AbstractReceiverWithQueue.run(AbstractReceiverWithQueue.java:172)
java.lang.Thread.run(Thread.java:808)

The current writer is:
Thread[Default background executor pool-2-thread-9,4,main]
com.sap.sailing.domain.tracking.AbstractTrackedRegattaAndRaceObserver$RegattaListener.raceRemoved(AbstractTrackedRegattaAndRaceObserver.java:101)
com.sap.sailing.domain.tracking.AbstractTrackedRegattaAndRaceObserver$RegattaListener$1.raceRemoved(AbstractTrackedRegattaAndRaceObserver.java:89)
com.sap.sailing.domain.tracking.impl.TrackedRegattaImpl.removeTrackedRace(TrackedRegattaImpl.java:139)
com.sap.sailing.server.impl.RacingEventServiceImpl.removeRace(RacingEventServiceImpl.java:2439)
com.sap.sailing.domain.tractracadapter.impl.TracTracRaceTrackerImpl.<init>(TracTracRaceTrackerImpl.java:398)
com.sap.sailing.domain.tractracadapter.impl.TracTracRaceTrackerImpl.<init>(TracTracRaceTrackerImpl.java:267)
com.sap.sailing.domain.tractracadapter.impl.DomainFactoryImpl.createRaceTracker(DomainFactoryImpl.java:710)
com.sap.sailing.domain.tractracadapter.impl.RaceTrackingConnectivityParametersImpl.createRaceTracker(RaceTrackingConnectivityParametersImpl.java:120)
com.sap.sailing.server.impl.RacingEventServiceImpl.addRace(RacingEventServiceImpl.java:1571)
com.sap.sailing.server.impl.RacingEventServiceImpl.lambda$4(RacingEventServiceImpl.java:791)
com.sap.sailing.server.impl.RacingEventServiceImpl$$Lambda$30/450464248.accept(Unknown Source)
com.sap.sailing.domain.persistence.impl.DomainObjectFactoryImpl.lambda$2(DomainObjectFactoryImpl.java:2381)
com.sap.sailing.domain.persistence.impl.DomainObjectFactoryImpl$$Lambda$39/2053027844.withService(Unknown Source)
com.sap.sse.osgi.CachedOsgiTypeBasedServiceFinder.applyServiceWhenAvailable(CachedOsgiTypeBasedServiceFinder.java:82)
com.sap.sailing.domain.persistence.impl.DomainObjectFactoryImpl.lambda$1(DomainObjectFactoryImpl.java:2370)
com.sap.sailing.domain.persistence.impl.DomainObjectFactoryImpl$$Lambda$38/270305695.run(Unknown Source)
java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
java.util.concurrent.FutureTask.run(FutureTask.java:266)
java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
java.util.concurrent.FutureTask.run(FutureTask.java:266)
java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
com.sap.sse.util.impl.ThreadPoolAwareRunnableScheduledFutureDelegate.run(ThreadPoolAwareRunnableScheduledFutureDelegate.java:23)
java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
java.lang.Thread.run(Thread.java:808)

Trying again...
Jan 13, 2018 3:49:10 PM com.sap.sse.concurrent.LockUtil lock
INFO: Couldn't acquire lock readLock trackeRaces lock for tracked regatta SCL 2017 - Porto Cervo (3) in 5s in thread com.sap.sailing.domain.tractracadapter.impl.RaceCourseReceiver at java.lang.Thread.getStackTrace(Thread.java:1628)
com.sap.sse.concurrent.LockUtil.getCurrentStackTrace(LockUtil.java:371)
com.sap.sse.concurrent.LockUtil.lock(LockUtil.java:401)
com.sap.sse.concurrent.LockUtil.acquireLockVirtuallyOrActually(LockUtil.java:114)
com.sap.sse.concurrent.LockUtil.lockForRead(LockUtil.java:92)
com.sap.sailing.domain.tracking.impl.TrackedRegattaImpl.lockTrackedRacesForRead(TrackedRegattaImpl.java:68)
com.sap.sailing.domain.tracking.impl.TrackedRegattaImpl.getExistingTrackedRace(TrackedRegattaImpl.java:198)
com.sap.sailing.domain.tracking.impl.DynamicTrackedRegattaImpl.getExistingTrackedRace(DynamicTrackedRegattaImpl.java:32)
com.sap.sailing.domain.tracking.impl.DynamicTrackedRegattaImpl.getExistingTrackedRace(DynamicTrackedRegattaImpl.java:1)
com.sap.sailing.domain.tracking.impl.TrackedRegattaImpl.getTrackedRace(TrackedRegattaImpl.java:162)
com.sap.sailing.domain.tracking.impl.DynamicTrackedRegattaImpl.getTrackedRace(DynamicTrackedRegattaImpl.java:21)
com.sap.sailing.domain.tracking.AbstractRaceTrackerImpl.getRaceIdentifier(AbstractRaceTrackerImpl.java:18)
com.sap.sailing.server.impl.RacingEventServiceImpl.getRaceTrackerByRegattaAndRaceIdentifier(RacingEventServiceImpl.java:3917)
com.sap.sailing.domain.racelogtracking.impl.fixtracker.RegattaLogFixTrackerRegattaListener.onRaceAdded(RegattaLogFixTrackerRegattaListener.java:73)
com.sap.sailing.domain.tracking.AbstractTrackedRegattaAndRaceObserver$RegattaListener.raceAdded(AbstractTrackedRegattaAndRaceObserver.java:112)
com.sap.sailing.domain.tracking.AbstractTrackedRegattaAndRaceObserver$RegattaListener$1.raceAdded(AbstractTrackedRegattaAndRaceObserver.java:94)
com.sap.sailing.domain.tracking.impl.TrackedRegattaImpl.addTrackedRace(TrackedRegattaImpl.java:118)
com.sap.sailing.domain.tracking.impl.TrackedRegattaImpl.createTrackedRace(TrackedRegattaImpl.java:260)
com.sap.sailing.domain.tracking.impl.DynamicTrackedRegattaImpl.createTrackedRace(DynamicTrackedRegattaImpl.java:39)
com.sap.sailing.domain.tractracadapter.impl.DomainFactoryImpl.createTrackedRace(DomainFactoryImpl.java:611)
com.sap.sailing.domain.tractracadapter.impl.DomainFactoryImpl.getOrCreateRaceDefinitionAndTrackedRace(DomainFactoryImpl.java:518)
com.sap.sailing.domain.tractracadapter.impl.RaceCourseReceiver.handleEvent(RaceCourseReceiver.java:162)
com.sap.sailing.domain.tractracadapter.impl.AbstractReceiverWithQueue.run(AbstractReceiverWithQueue.java:172)
java.lang.Thread.run(Thread.java:808)

The current writer is:
 Thread[Default background executor pool-2-thread-9,4,main]
com.sap.sailing.domain.tracking.AbstractTrackedRegattaAndRaceObserver$RegattaListener.raceRemoved(AbstractTrackedRegattaAndRaceObserver.java:101)
com.sap.sailing.domain.tracking.AbstractTrackedRegattaAndRaceObserver$RegattaListener$1.raceRemoved(AbstractTrackedRegattaAndRaceObserver.java:89)
com.sap.sailing.domain.tracking.impl.TrackedRegattaImpl.removeTrackedRace(TrackedRegattaImpl.java:139)
com.sap.sailing.server.impl.RacingEventServiceImpl.removeRace(RacingEventServiceImpl.java:2439)
com.sap.sailing.domain.tractracadapter.impl.TracTracRaceTrackerImpl.<init>(TracTracRaceTrackerImpl.java:398)
com.sap.sailing.domain.tractracadapter.impl.TracTracRaceTrackerImpl.<init>(TracTracRaceTrackerImpl.java:267)
com.sap.sailing.domain.tractracadapter.impl.DomainFactoryImpl.createRaceTracker(DomainFactoryImpl.java:710)
com.sap.sailing.domain.tractracadapter.impl.RaceTrackingConnectivityParametersImpl.createRaceTracker(RaceTrackingConnectivityParametersImpl.java:120)
com.sap.sailing.server.impl.RacingEventServiceImpl.addRace(RacingEventServiceImpl.java:1571)
com.sap.sailing.server.impl.RacingEventServiceImpl.lambda$4(RacingEventServiceImpl.java:791)
com.sap.sailing.server.impl.RacingEventServiceImpl$$Lambda$30/450464248.accept(Unknown Source)
com.sap.sailing.domain.persistence.impl.DomainObjectFactoryImpl.lambda$2(DomainObjectFactoryImpl.java:2381)
com.sap.sailing.domain.persistence.impl.DomainObjectFactoryImpl$$Lambda$39/2053027844.withService(Unknown Source)
com.sap.sse.osgi.CachedOsgiTypeBasedServiceFinder.applyServiceWhenAvailable(CachedOsgiTypeBasedServiceFinder.java:82)
com.sap.sailing.domain.persistence.impl.DomainObjectFactoryImpl.lambda$1(DomainObjectFactoryImpl.java:2370)
com.sap.sailing.domain.persistence.impl.DomainObjectFactoryImpl$$Lambda$38/270305695.run(Unknown Source)
java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
java.util.concurrent.FutureTask.run(FutureTask.java:266)
java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
java.util.concurrent.FutureTask.run(FutureTask.java:266)
java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
com.sap.sse.util.impl.ThreadPoolAwareRunnableScheduledFutureDelegate.run(ThreadPoolAwareRunnableScheduledFutureDelegate.java:23)
java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
java.lang.Thread.run(Thread.java:808)

Trying again...
Jan 13, 2018 3:49:10 PM com.sap.sse.concurrent.LockUtil lock
INFO: Couldn't acquire lock readLock trackeRaces lock for tracked regatta SCL 2017 - Porto Cervo (3) in 5s in thread qtp708032164-166433 at java.lang.Thread.getStackTrace(Thread.java:1628)
com.sap.sse.concurrent.LockUtil.getCurrentStackTrace(LockUtil.java:371)
com.sap.sse.concurrent.LockUtil.lock(LockUtil.java:401)
com.sap.sse.concurrent.LockUtil.acquireLockVirtuallyOrActually(LockUtil.java:114)
com.sap.sse.concurrent.LockUtil.lockForRead(LockUtil.java:92)
com.sap.sailing.domain.tracking.impl.TrackedRegattaImpl.lockTrackedRacesForRead(TrackedRegattaImpl.java:68)
com.sap.sailing.domain.tracking.impl.TrackedRegattaImpl.getExistingTrackedRace(TrackedRegattaImpl.java:198)
com.sap.sailing.domain.tracking.impl.DynamicTrackedRegattaImpl.getExistingTrackedRace(DynamicTrackedRegattaImpl.java:32)
com.sap.sailing.server.impl.RacingEventServiceImpl.getExistingTrackedRace(RacingEventServiceImpl.java:2641)
com.sap.sailing.gwt.ui.server.SailingServiceImpl.convertToRaceDTOs(SailingServiceImpl.java:1081)
com.sap.sailing.gwt.ui.server.SailingServiceImpl.convertToRegattaDTO(SailingServiceImpl.java:902)
com.sap.sailing.gwt.ui.server.SailingServiceImpl.getRegattas(SailingServiceImpl.java:869)
sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
 sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
java.lang.reflect.Method.invoke(Method.java:498)
com.google.gwt.user.server.rpc.RPC.invokeAndEncodeResponse(RPC.java:587)
com.google.gwt.user.server.rpc.RemoteServiceServlet.processCall(RemoteServiceServlet.java:333)
com.google.gwt.user.server.rpc.RemoteServiceServlet.processCall(RemoteServiceServlet.java:303)
com.google.gwt.user.server.rpc.RemoteServiceServlet.processPost(RemoteServiceServlet.java:373)
com.google.gwt.user.server.rpc.AbstractRemoteServiceServlet.doPost(AbstractRemoteServiceServlet.java:62)
javax.servlet.http.HttpServlet.service(HttpServlet.java:707)
javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:769)
org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1667)
com.sap.sse.gwt.server.GWTCacheControlServletFilter.doFilter(GWTCacheControlServletFilter.java:44)
org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1650)
org.apache.shiro.web.servlet.AbstractShiroFilter.executeChain(AbstractShiroFilter.java:449)
org.apache.shiro.web.servlet.AbstractShiroFilter$1.call(AbstractShiroFilter.java:365)
org.apache.shiro.subject.support.SubjectCallable.doCall(SubjectCallable.java:90)
org.apache.shiro.subject.support.SubjectCallable.call(SubjectCallable.java:83)
org.apache.shiro.subject.support.DelegatingSubject.execute(DelegatingSubject.java:383)
org.apache.shiro.web.servlet.AbstractShiroFilter.doFilterInternal(AbstractShiroFilter.java:362)
org.apache.shiro.web.servlet.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:125)
org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1650)
org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:583)
org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:553)
org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:223)
org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1125)
org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:515)
org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:185)
org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1059)
org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:215)
org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:110)
org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:97)
org.eclipse.jetty.server.Server.handle(Server.java:485)
org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:290)
org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:248)
org.eclipse.jetty.io.AbstractConnection$2.run(AbstractConnection.java:540)
org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:606)
org.eclipse.jetty.util.thread.QueuedThreadPool$3.run(QueuedThreadPool.java:535)
java.lang.Thread.run(Thread.java:808)

The current writer is:
Thread[Default background executor pool-2-thread-9,4,main]
com.sap.sailing.domain.tracking.AbstractTrackedRegattaAndRaceObserver$RegattaListener.raceRemoved(AbstractTrackedRegattaAndRaceObserver.java:101)
com.sap.sailing.domain.tracking.AbstractTrackedRegattaAndRaceObserver$RegattaListener$1.raceRemoved(AbstractTrackedRegattaAndRaceObserver.java:89)
com.sap.sailing.domain.tracking.impl.TrackedRegattaImpl.removeTrackedRace(TrackedRegattaImpl.java:139)
com.sap.sailing.server.impl.RacingEventServiceImpl.removeRace(RacingEventServiceImpl.java:2439)
com.sap.sailing.domain.tractracadapter.impl.TracTracRaceTrackerImpl.<init>(TracTracRaceTrackerImpl.java:398)
com.sap.sailing.domain.tractracadapter.impl.TracTracRaceTrackerImpl.<init>(TracTracRaceTrackerImpl.java:267)
com.sap.sailing.domain.tractracadapter.impl.DomainFactoryImpl.createRaceTracker(DomainFactoryImpl.java:710)
com.sap.sailing.domain.tractracadapter.impl.RaceTrackingConnectivityParametersImpl.createRaceTracker(RaceTrackingConnectivityParametersImpl.java:120)
com.sap.sailing.server.impl.RacingEventServiceImpl.addRace(RacingEventServiceImpl.java:1571)
com.sap.sailing.server.impl.RacingEventServiceImpl.lambda$4(RacingEventServiceImpl.java:791)
com.sap.sailing.server.impl.RacingEventServiceImpl$$Lambda$30/450464248.accept(Unknown Source)
com.sap.sailing.domain.persistence.impl.DomainObjectFactoryImpl.lambda$2(DomainObjectFactoryImpl.java:2381)
com.sap.sailing.domain.persistence.impl.DomainObjectFactoryImpl$$Lambda$39/2053027844.withService(Unknown Source)
com.sap.sse.osgi.CachedOsgiTypeBasedServiceFinder.applyServiceWhenAvailable(CachedOsgiTypeBasedServiceFinder.java:82)
com.sap.sailing.domain.persistence.impl.DomainObjectFactoryImpl.lambda$1(DomainObjectFactoryImpl.java:2370)
com.sap.sailing.domain.persistence.impl.DomainObjectFactoryImpl$$Lambda$38/270305695.run(Unknown Source)
java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
java.util.concurrent.FutureTask.run(FutureTask.java:266)
java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
java.util.concurrent.FutureTask.run(FutureTask.java:266)
java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
com.sap.sse.util.impl.ThreadPoolAwareRunnableScheduledFutureDelegate.run(ThreadPoolAwareRunnableScheduledFutureDelegate.java:23)
java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
java.lang.Thread.run(Thread.java:808)

Trying again...
Jan 13, 2018 3:49:10 PM com.sap.sse.concurrent.LockUtil lock
INFO: Couldn't acquire lock writeLock raceTrackersByRegatta for RacingEventService: 2145162434 Build: e0a6d0ac2fa874f51b9c68d58e9c7d10f1fecd93-build-201801131158  System: dbserver.internal.sapsailing.com:10202/winddb-2010-rabbit.internal.sapsailing.com:0/sapsailinganalytics-archive Started: 201801131208 in 5s in thread Default background executor pool-2-thread-22 at java.lang.Thread.getStackTrace(Thread.java:1628)
com.sap.sse.concurrent.LockUtil.getCurrentStackTrace(LockUtil.java:371)
com.sap.sse.concurrent.LockUtil.lock(LockUtil.java:401)
com.sap.sse.concurrent.LockUtil.acquireLockVirtuallyOrActually(LockUtil.java:114)
com.sap.sse.concurrent.LockUtil.lockForWrite(LockUtil.java:96)
com.sap.sailing.server.impl.RacingEventServiceImpl.addRace(RacingEventServiceImpl.java:1577)
com.sap.sailing.server.impl.RacingEventServiceImpl.lambda$4(RacingEventServiceImpl.java:791)
com.sap.sailing.domain.persistence.impl.DomainObjectFactoryImpl.lambda$2(DomainObjectFactoryImpl.java:2381)
com.sap.sse.osgi.CachedOsgiTypeBasedServiceFinder.applyServiceWhenAvailable(CachedOsgiTypeBasedServiceFinder.java:82)
com.sap.sailing.domain.persistence.impl.DomainObjectFactoryImpl.lambda$1(DomainObjectFactoryImpl.java:2370)
java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
java.util.concurrent.FutureTask.run(FutureTask.java:266)
java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
java.util.concurrent.FutureTask.run(FutureTask.java:266)
java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
com.sap.sse.util.impl.ThreadPoolAwareRunnableScheduledFutureDelegate.run(ThreadPoolAwareRunnableScheduledFutureDelegate.java:23)
java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
java.lang.Thread.run(Thread.java:808)

The current readers are:
Thread[com.sap.sailing.domain.tractracadapter.impl.RaceCourseReceiver,4,main]
java.util.logging.FileHandler.publish(FileHandler.java:679)
java.util.logging.Logger.log(Logger.java:738)
java.util.logging.Logger.doLog(Logger.java:765)
java.util.logging.Logger.log(Logger.java:788)
java.util.logging.Logger.info(Logger.java:1490)
com.sap.sse.concurrent.LockUtil.lock(LockUtil.java:413)
com.sap.sse.concurrent.LockUtil.acquireLockVirtuallyOrActually(LockUtil.java:114)
com.sap.sse.concurrent.LockUtil.lockForRead(LockUtil.java:92)
com.sap.sailing.domain.tracking.impl.TrackedRegattaImpl.lockTrackedRacesForRead(TrackedRegattaImpl.java:68)
com.sap.sailing.domain.tracking.impl.TrackedRegattaImpl.getExistingTrackedRace(TrackedRegattaImpl.java:198)
com.sap.sailing.domain.tracking.impl.DynamicTrackedRegattaImpl.getExistingTrackedRace(DynamicTrackedRegattaImpl.java:32)
com.sap.sailing.domain.tracking.impl.DynamicTrackedRegattaImpl.getExistingTrackedRace(DynamicTrackedRegattaImpl.java:1)
com.sap.sailing.domain.tracking.impl.TrackedRegattaImpl.getTrackedRace(TrackedRegattaImpl.java:162)
com.sap.sailing.domain.tracking.impl.DynamicTrackedRegattaImpl.getTrackedRace(DynamicTrackedRegattaImpl.java:21)
com.sap.sailing.domain.tracking.AbstractRaceTrackerImpl.getRaceIdentifier(AbstractRaceTrackerImpl.java:18)
com.sap.sailing.server.impl.RacingEventServiceImpl.getRaceTrackerByRegattaAndRaceIdentifier(RacingEventServiceImpl.java:3917)
com.sap.sailing.domain.racelogtracking.impl.fixtracker.RegattaLogFixTrackerRegattaListener.onRaceAdded(RegattaLogFixTrackerRegattaListener.java:73)
com.sap.sailing.domain.tracking.AbstractTrackedRegattaAndRaceObserver$RegattaListener.raceAdded(AbstractTrackedRegattaAndRaceObserver.java:112)
com.sap.sailing.domain.tracking.AbstractTrackedRegattaAndRaceObserver$RegattaListener$1.raceAdded(AbstractTrackedRegattaAndRaceObserver.java:94)
com.sap.sailing.domain.tracking.impl.TrackedRegattaImpl.addTrackedRace(TrackedRegattaImpl.java:118)
com.sap.sailing.domain.tracking.impl.TrackedRegattaImpl.createTrackedRace(TrackedRegattaImpl.java:260)
com.sap.sailing.domain.tracking.impl.DynamicTrackedRegattaImpl.createTrackedRace(DynamicTrackedRegattaImpl.java:39)
com.sap.sailing.domain.tractracadapter.impl.DomainFactoryImpl.createTrackedRace(DomainFactoryImpl.java:611)
com.sap.sailing.domain.tractracadapter.impl.DomainFactoryImpl.getOrCreateRaceDefinitionAndTrackedRace(DomainFactoryImpl.java:518)
com.sap.sailing.domain.tractracadapter.impl.RaceCourseReceiver.handleEvent(RaceCourseReceiver.java:162)
com.sap.sailing.domain.tractracadapter.impl.AbstractReceiverWithQueue.run(AbstractReceiverWithQueue.java:172)
java.lang.Thread.run(Thread.java:808)

Trying again...
Jan 13, 2018 3:49:10 PM com.sap.sse.concurrent.LockUtil lock
INFO: Couldn't acquire lock readLock trackeRaces lock for tracked regatta SCL 2017 - Porto Cervo (3) in 5s in thread qtp708032164-166505 at java.lang.Thread.getStackTrace(Thread.java:1628)
com.sap.sse.concurrent.LockUtil.getCurrentStackTrace(LockUtil.java:371)
com.sap.sse.concurrent.LockUtil.lock(LockUtil.java:401)
com.sap.sse.concurrent.LockUtil.acquireLockVirtuallyOrActually(LockUtil.java:114)
com.sap.sse.concurrent.LockUtil.lockForRead(LockUtil.java:92)
com.sap.sailing.domain.tracking.impl.TrackedRegattaImpl.lockTrackedRacesForRead(TrackedRegattaImpl.java:68)
com.sap.sailing.domain.tracking.impl.TrackedRegattaImpl.getExistingTrackedRace(TrackedRegattaImpl.java:198)
com.sap.sailing.domain.tracking.impl.DynamicTrackedRegattaImpl.getExistingTrackedRace(DynamicTrackedRegattaImpl.java:32)
com.sap.sailing.server.impl.RacingEventServiceImpl.getExistingTrackedRace(RacingEventServiceImpl.java:2641)
com.sap.sailing.gwt.ui.server.SailingServiceImpl.convertToRaceDTOs(SailingServiceImpl.java:1081)
com.sap.sailing.gwt.ui.server.SailingServiceImpl.convertToRegattaDTO(SailingServiceImpl.java:902)
com.sap.sailing.gwt.ui.server.SailingServiceImpl.getRegattas(SailingServiceImpl.java:869)
sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
java.lang.reflect.Method.invoke(Method.java:498)
com.google.gwt.user.server.rpc.RPC.invokeAndEncodeResponse(RPC.java:587)
com.google.gwt.user.server.rpc.RemoteServiceServlet.processCall(RemoteServiceServlet.java:333)
com.google.gwt.user.server.rpc.RemoteServiceServlet.processCall(RemoteServiceServlet.java:303)
com.google.gwt.user.server.rpc.RemoteServiceServlet.processPost(RemoteServiceServlet.java:373)
com.google.gwt.user.server.rpc.AbstractRemoteServiceServlet.doPost(AbstractRemoteServiceServlet.java:62)
javax.servlet.http.HttpServlet.service(HttpServlet.java:707)
javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:769)
org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1667)
com.sap.sse.gwt.server.GWTCacheControlServletFilter.doFilter(GWTCacheControlServletFilter.java:44)
org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1650)
org.apache.shiro.web.servlet.AbstractShiroFilter.executeChain(AbstractShiroFilter.java:449)
org.apache.shiro.web.servlet.AbstractShiroFilter$1.call(AbstractShiroFilter.java:365)
org.apache.shiro.subject.support.SubjectCallable.doCall(SubjectCallable.java:90)
org.apache.shiro.subject.support.SubjectCallable.call(SubjectCallable.java:83)
org.apache.shiro.subject.support.DelegatingSubject.execute(DelegatingSubject.java:383)
org.apache.shiro.web.servlet.AbstractShiroFilter.doFilterInternal(AbstractShiroFilter.java:362)
org.apache.shiro.web.servlet.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:125)
org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1650)
org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:583)
org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:553)
org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:223)
org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1125)
org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:515)
org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:185)
org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1059)
org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:215)
org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:110)
org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:97)
org.eclipse.jetty.server.Server.handle(Server.java:485)
org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:290)
org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:248)
org.eclipse.jetty.io.AbstractConnection$2.run(AbstractConnection.java:540)
org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:606)
org.eclipse.jetty.util.thread.QueuedThreadPool$3.run(QueuedThreadPool.java:535)
java.lang.Thread.run(Thread.java:808)

The current writer is:
Thread[Default background executor pool-2-thread-9,4,main]
com.sap.sailing.domain.tracking.AbstractTrackedRegattaAndRaceObserver$RegattaListener.raceRemoved(AbstractTrackedRegattaAndRaceObserver.java:101)
com.sap.sailing.domain.tracking.AbstractTrackedRegattaAndRaceObserver$RegattaListener$1.raceRemoved(AbstractTrackedRegattaAndRaceObserver.java:89)
com.sap.sailing.domain.tracking.impl.TrackedRegattaImpl.removeTrackedRace(TrackedRegattaImpl.java:139)
com.sap.sailing.server.impl.RacingEventServiceImpl.removeRace(RacingEventServiceImpl.java:2439)
com.sap.sailing.domain.tractracadapter.impl.TracTracRaceTrackerImpl.<init>(TracTracRaceTrackerImpl.java:398)
com.sap.sailing.domain.tractracadapter.impl.TracTracRaceTrackerImpl.<init>(TracTracRaceTrackerImpl.java:267)
com.sap.sailing.domain.tractracadapter.impl.DomainFactoryImpl.createRaceTracker(DomainFactoryImpl.java:710)
com.sap.sailing.domain.tractracadapter.impl.RaceTrackingConnectivityParametersImpl.createRaceTracker(RaceTrackingConnectivityParametersImpl.java:120)
com.sap.sailing.server.impl.RacingEventServiceImpl.addRace(RacingEventServiceImpl.java:1571)
com.sap.sailing.server.impl.RacingEventServiceImpl.lambda$4(RacingEventServiceImpl.java:791)
com.sap.sailing.server.impl.RacingEventServiceImpl$$Lambda$30/450464248.accept(Unknown Source)
com.sap.sailing.domain.persistence.impl.DomainObjectFactoryImpl.lambda$2(DomainObjectFactoryImpl.java:2381)
com.sap.sailing.domain.persistence.impl.DomainObjectFactoryImpl$$Lambda$39/2053027844.withService(Unknown Source)
com.sap.sse.osgi.CachedOsgiTypeBasedServiceFinder.applyServiceWhenAvailable(CachedOsgiTypeBasedServiceFinder.java:82)
com.sap.sailing.domain.persistence.impl.DomainObjectFactoryImpl.lambda$1(DomainObjectFactoryImpl.java:2370)
com.sap.sailing.domain.persistence.impl.DomainObjectFactoryImpl$$Lambda$38/270305695.run(Unknown Source)
java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
java.util.concurrent.FutureTask.run(FutureTask.java:266)
java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
java.util.concurrent.FutureTask.run(FutureTask.java:266)
java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
com.sap.sse.util.impl.ThreadPoolAwareRunnableScheduledFutureDelegate.run(ThreadPoolAwareRunnableScheduledFutureDelegate.java:23)
java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
java.lang.Thread.run(Thread.java:808)

Trying again...
Jan 13, 2018 3:49:10 PM com.sap.sse.concurrent.LockUtil lock
INFO: Couldn't acquire lock readLock trackeRaces lock for tracked regatta SCL 2017 - Porto Cervo (3) in 5s in thread com.sap.sailing.domain.tractracadapter.impl.RawPositionReceiver at java.lang.Thread.getStackTrace(Thread.java:1628)
com.sap.sse.concurrent.LockUtil.getCurrentStackTrace(LockUtil.java:371)
com.sap.sse.concurrent.LockUtil.lock(LockUtil.java:401)
com.sap.sse.concurrent.LockUtil.acquireLockVirtuallyOrActually(LockUtil.java:114)
com.sap.sse.concurrent.LockUtil.lockForRead(LockUtil.java:92)
com.sap.sailing.domain.tracking.impl.TrackedRegattaImpl.lockTrackedRacesForRead(TrackedRegattaImpl.java:68)
com.sap.sailing.domain.tracking.impl.TrackedRegattaImpl.getExistingTrackedRace(TrackedRegattaImpl.java:198)
com.sap.sailing.domain.tracking.impl.DynamicTrackedRegattaImpl.getExistingTrackedRace(DynamicTrackedRegattaImpl.java:32)
com.sap.sailing.domain.tracking.impl.DynamicTrackedRegattaImpl.getExistingTrackedRace(DynamicTrackedRegattaImpl.java:1)
com.sap.sailing.domain.tracking.impl.TrackedRegattaImpl.getTrackedRace(TrackedRegattaImpl.java:162)
com.sap.sailing.domain.tracking.impl.DynamicTrackedRegattaImpl.getTrackedRace(DynamicTrackedRegattaImpl.java:21)
com.sap.sailing.domain.tractracadapter.impl.AbstractReceiverWithQueue.getTrackedRace(AbstractReceiverWithQueue.java:237)
com.sap.sailing.domain.tractracadapter.impl.RawPositionReceiver.handleEvent(RawPositionReceiver.java:59)
com.sap.sailing.domain.tractracadapter.impl.AbstractReceiverWithQueue.run(AbstractReceiverWithQueue.java:172)
java.lang.Thread.run(Thread.java:808)

The current writer is:
Thread[Default background executor pool-2-thread-9,4,main]
com.sap.sailing.domain.tracking.AbstractTrackedRegattaAndRaceObserver$RegattaListener.raceRemoved(AbstractTrackedRegattaAndRaceObserver.java:101)
com.sap.sailing.domain.tracking.AbstractTrackedRegattaAndRaceObserver$RegattaListener$1.raceRemoved(AbstractTrackedRegattaAndRaceObserver.java:89)
com.sap.sailing.domain.tracking.impl.TrackedRegattaImpl.removeTrackedRace(TrackedRegattaImpl.java:139)
com.sap.sailing.server.impl.RacingEventServiceImpl.removeRace(RacingEventServiceImpl.java:2439)
com.sap.sailing.domain.tractracadapter.impl.TracTracRaceTrackerImpl.<init>(TracTracRaceTrackerImpl.java:398)
com.sap.sailing.domain.tractracadapter.impl.TracTracRaceTrackerImpl.<init>(TracTracRaceTrackerImpl.java:267)
com.sap.sailing.domain.tractracadapter.impl.DomainFactoryImpl.createRaceTracker(DomainFactoryImpl.java:710)
com.sap.sailing.domain.tractracadapter.impl.RaceTrackingConnectivityParametersImpl.createRaceTracker(RaceTrackingConnectivityParametersImpl.java:120)
com.sap.sailing.server.impl.RacingEventServiceImpl.addRace(RacingEventServiceImpl.java:1571)
com.sap.sailing.server.impl.RacingEventServiceImpl.lambda$4(RacingEventServiceImpl.java:791)
com.sap.sailing.server.impl.RacingEventServiceImpl$$Lambda$30/450464248.accept(Unknown Source)
com.sap.sailing.domain.persistence.impl.DomainObjectFactoryImpl.lambda$2(DomainObjectFactoryImpl.java:2381)
com.sap.sailing.domain.persistence.impl.DomainObjectFactoryImpl$$Lambda$39/2053027844.withService(Unknown Source)
com.sap.sse.osgi.CachedOsgiTypeBasedServiceFinder.applyServiceWhenAvailable(CachedOsgiTypeBasedServiceFinder.java:82)
com.sap.sailing.domain.persistence.impl.DomainObjectFactoryImpl.lambda$1(DomainObjectFactoryImpl.java:2370)
com.sap.sailing.domain.persistence.impl.DomainObjectFactoryImpl$$Lambda$38/270305695.run(Unknown Source)
java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
java.util.concurrent.FutureTask.run(FutureTask.java:266)
java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
java.util.concurrent.FutureTask.run(FutureTask.java:266)
java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
com.sap.sse.util.impl.ThreadPoolAwareRunnableScheduledFutureDelegate.run(ThreadPoolAwareRunnableScheduledFutureDelegate.java:23)
java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
java.lang.Thread.run(Thread.java:808)

Trying again...
Jan 13, 2018 3:49:10 PM com.sap.sse.concurrent.LockUtil lock
INFO: Couldn't acquire lock readLock trackeRaces lock for tracked regatta SCL 2017 - Porto Cervo (3) in 5s in thread qtp708032164-166572 at java.lang.Thread.getStackTrace(Thread.java:1628)
com.sap.sse.concurrent.LockUtil.getCurrentStackTrace(LockUtil.java:371)
com.sap.sse.concurrent.LockUtil.lock(LockUtil.java:401)
com.sap.sse.concurrent.LockUtil.acquireLockVirtuallyOrActually(LockUtil.java:114)
com.sap.sse.concurrent.LockUtil.lockForRead(LockUtil.java:92)
com.sap.sailing.domain.tracking.impl.TrackedRegattaImpl.lockTrackedRacesForRead(TrackedRegattaImpl.java:68)
com.sap.sailing.domain.tracking.impl.TrackedRegattaImpl.getExistingTrackedRace(TrackedRegattaImpl.java:198)
com.sap.sailing.domain.tracking.impl.DynamicTrackedRegattaImpl.getExistingTrackedRace(DynamicTrackedRegattaImpl.java:32)
com.sap.sailing.server.impl.RacingEventServiceImpl.getExistingTrackedRace(RacingEventServiceImpl.java:2641)
com.sap.sailing.gwt.ui.server.SailingServiceImpl.convertToRaceDTOs(SailingServiceImpl.java:1081)
com.sap.sailing.gwt.ui.server.SailingServiceImpl.convertToRegattaDTO(SailingServiceImpl.java:902)
com.sap.sailing.gwt.ui.server.SailingServiceImpl.getRegattas(SailingServiceImpl.java:869)
sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
java.lang.reflect.Method.invoke(Method.java:498)
com.google.gwt.user.server.rpc.RPC.invokeAndEncodeResponse(RPC.java:587)
com.google.gwt.user.server.rpc.RemoteServiceServlet.processCall(RemoteServiceServlet.java:333)
com.google.gwt.user.server.rpc.RemoteServiceServlet.processCall(RemoteServiceServlet.java:303)
com.google.gwt.user.server.rpc.RemoteServiceServlet.processPost(RemoteServiceServlet.java:373)
com.google.gwt.user.server.rpc.AbstractRemoteServiceServlet.doPost(AbstractRemoteServiceServlet.java:62)
javax.servlet.http.HttpServlet.service(HttpServlet.java:707)
javax.servlet.http.HttpServlet.service(HttpServlet.java:790)
org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:769)
org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1667)
com.sap.sse.gwt.server.GWTCacheControlServletFilter.doFilter(GWTCacheControlServletFilter.java:44)
org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1650)
org.apache.shiro.web.servlet.AbstractShiroFilter.executeChain(AbstractShiroFilter.java:449)
org.apache.shiro.web.servlet.AbstractShiroFilter$1.call(AbstractShiroFilter.java:365)
org.apache.shiro.subject.support.SubjectCallable.doCall(SubjectCallable.java:90)
org.apache.shiro.subject.support.SubjectCallable.call(SubjectCallable.java:83)
org.apache.shiro.subject.support.DelegatingSubject.execute(DelegatingSubject.java:383)
org.apache.shiro.web.servlet.AbstractShiroFilter.doFilterInternal(AbstractShiroFilter.java:362)
org.apache.shiro.web.servlet.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:125)
org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1650)
org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:583)
org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:553)
org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:223)
org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1125)
org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:515)
org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:185)
org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1059)
org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:215)
org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:110)
org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:97)
org.eclipse.jetty.server.Server.handle(Server.java:485)
org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:290)
org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:248)
org.eclipse.jetty.io.AbstractConnection$2.run(AbstractConnection.java:540)
org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:606)
org.eclipse.jetty.util.thread.QueuedThreadPool$3.run(QueuedThreadPool.java:535)
java.lang.Thread.run(Thread.java:808)

The current writer is:
Thread[Default background executor pool-2-thread-9,4,main]
com.sap.sailing.domain.tracking.AbstractTrackedRegattaAndRaceObserver$RegattaListener.raceRemoved(AbstractTrackedRegattaAndRaceObserver.java:101)
com.sap.sailing.domain.tracking.AbstractTrackedRegattaAndRaceObserver$RegattaListener$1.raceRemoved(AbstractTrackedRegattaAndRaceObserver.java:89)
com.sap.sailing.domain.tracking.impl.TrackedRegattaImpl.removeTrackedRace(TrackedRegattaImpl.java:139)
com.sap.sailing.server.impl.RacingEventServiceImpl.removeRace(RacingEventServiceImpl.java:2439)
com.sap.sailing.domain.tractracadapter.impl.TracTracRaceTrackerImpl.<init>(TracTracRaceTrackerImpl.java:398)
com.sap.sailing.domain.tractracadapter.impl.TracTracRaceTrackerImpl.<init>(TracTracRaceTrackerImpl.java:267)
com.sap.sailing.domain.tractracadapter.impl.DomainFactoryImpl.createRaceTracker(DomainFactoryImpl.java:710)
com.sap.sailing.domain.tractracadapter.impl.RaceTrackingConnectivityParametersImpl.createRaceTracker(RaceTrackingConnectivityParametersImpl.java:120)
com.sap.sailing.server.impl.RacingEventServiceImpl.addRace(RacingEventServiceImpl.java:1571)
com.sap.sailing.server.impl.RacingEventServiceImpl.lambda$4(RacingEventServiceImpl.java:791)
com.sap.sailing.server.impl.RacingEventServiceImpl$$Lambda$30/450464248.accept(Unknown Source)
com.sap.sailing.domain.persistence.impl.DomainObjectFactoryImpl.lambda$2(DomainObjectFactoryImpl.java:2381)
com.sap.sailing.domain.persistence.impl.DomainObjectFactoryImpl$$Lambda$39/2053027844.withService(Unknown Source)
com.sap.sse.osgi.CachedOsgiTypeBasedServiceFinder.applyServiceWhenAvailable(CachedOsgiTypeBasedServiceFinder.java:82)
com.sap.sailing.domain.persistence.impl.DomainObjectFactoryImpl.lambda$1(DomainObjectFactoryImpl.java:2370)
com.sap.sailing.domain.persistence.impl.DomainObjectFactoryImpl$$Lambda$38/270305695.run(Unknown Source)
java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
java.util.concurrent.FutureTask.run(FutureTask.java:266)
java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
java.util.concurrent.FutureTask.run(FutureTask.java:266)
java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
com.sap.sse.util.impl.ThreadPoolAwareRunnableScheduledFutureDelegate.run(ThreadPoolAwareRunnableScheduledFutureDelegate.java:23)
java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
java.lang.Thread.run(Thread.java:808)

Trying again...

**Comment by eclipse-sailing-analytics-bot** [comment]
**@axeluhl** commented:

Analysis:

542be865f8fa5c395f85f2f3ccc95b523711750c has introduced synchronization on AbstractTrackedRegattaAndRaceObserver.RegattaListener.raceRemoved(...). It seems that the thoughts that led to the commit's comment "added synchronized to AbstractTrackedRegattaAndRaceObserver.raceRemoved(); run SmartFutureCache.get() and remove() in background thread to avoid deadlock" were not entirely correct. The writer thread Thread[Default background executor pool-2-thread-9,4,main] is blocked on this method entry. The thread com.sap.sailing.domain.tractracadapter.impl.RaceCourseReceiver is interesting: while holding the RegattaListener's monitor in raceAdded(...) it tries to obtain the read lock:

	com.sap.sailing.domain.tracking.impl.TrackedRegattaImpl.lockTrackedRacesForRead(TrackedRegattaImpl.java:68)

Here is the stack trace excerpt:

com.sap.sse.concurrent.LockUtil.getCurrentStackTrace(LockUtil.java:371)
com.sap.sse.concurrent.LockUtil.lock(LockUtil.java:401)
com.sap.sse.concurrent.LockUtil.acquireLockVirtuallyOrActually(LockUtil.java:114)
com.sap.sse.concurrent.LockUtil.lockForRead(LockUtil.java:92)
com.sap.sailing.domain.tracking.impl.TrackedRegattaImpl.lockTrackedRacesForRead(TrackedRegattaImpl.java:68)
com.sap.sailing.domain.tracking.impl.TrackedRegattaImpl.getExistingTrackedRace(TrackedRegattaImpl.java:198)
com.sap.sailing.domain.tracking.impl.DynamicTrackedRegattaImpl.getExistingTrackedRace(DynamicTrackedRegattaImpl.java:32)
com.sap.sailing.domain.tracking.impl.DynamicTrackedRegattaImpl.getExistingTrackedRace(DynamicTrackedRegattaImpl.java:1)
com.sap.sailing.domain.tracking.impl.TrackedRegattaImpl.getTrackedRace(TrackedRegattaImpl.java:162)
com.sap.sailing.domain.tracking.impl.DynamicTrackedRegattaImpl.getTrackedRace(DynamicTrackedRegattaImpl.java:21)
com.sap.sailing.domain.tracking.AbstractRaceTrackerImpl.getRaceIdentifier(AbstractRaceTrackerImpl.java:18)
com.sap.sailing.server.impl.RacingEventServiceImpl.getRaceTrackerByRegattaAndRaceIdentifier(RacingEventServiceImpl.java:3917)
com.sap.sailing.domain.racelogtracking.impl.fixtracker.RegattaLogFixTrackerRegattaListener.onRaceAdded(RegattaLogFixTrackerRegattaListener.java:73)
com.sap.sailing.domain.tracking.AbstractTrackedRegattaAndRaceObserver$RegattaListener.raceAdded(AbstractTrackedRegattaAndRaceObserver.java:112)
com.sap.sailing.domain.tracking.AbstractTrackedRegattaAndRaceObserver$RegattaListener$1.raceAdded(AbstractTrackedRegattaAndRaceObserver.java:94)
com.sap.sailing.domain.tracking.impl.TrackedRegattaImpl.addTrackedRace(TrackedRegattaImpl.java:118)
com.sap.sailing.domain.tracking.impl.TrackedRegattaImpl.createTrackedRace(TrackedRegattaImpl.java:260)
com.sap.sailing.domain.tracking.impl.DynamicTrackedRegattaImpl.createTrackedRace(DynamicTrackedRegattaImpl.java:39)
com.sap.sailing.domain.tractracadapter.impl.DomainFactoryImpl.createTrackedRace(DomainFactoryImpl.java:611)
com.sap.sailing.domain.tractracadapter.impl.DomainFactoryImpl.getOrCreateRaceDefinitionAndTrackedRace(DomainFactoryImpl.java:518)
com.sap.sailing.domain.tractracadapter.impl.RaceCourseReceiver.handleEvent(RaceCourseReceiver.java:162)

The tracked races lock is being held by this thread:
Thread[Default background executor pool-2-thread-9,4,main]

com.sap.sailing.domain.tracking.AbstractTrackedRegattaAndRaceObserver$RegattaListener.raceRemoved(AbstractTrackedRegattaAndRaceObserver.java:101)
com.sap.sailing.domain.tracking.AbstractTrackedRegattaAndRaceObserver$RegattaListener$1.raceRemoved(AbstractTrackedRegattaAndRaceObserver.java:89)
com.sap.sailing.domain.tracking.impl.TrackedRegattaImpl.removeTrackedRace(TrackedRegattaImpl.java:139)
com.sap.sailing.server.impl.RacingEventServiceImpl.removeRace(RacingEventServiceImpl.java:2439)
com.sap.sailing.domain.tractracadapter.impl.TracTracRaceTrackerImpl.<init>(TracTracRaceTrackerImpl.java:398)
com.sap.sailing.domain.tractracadapter.impl.TracTracRaceTrackerImpl.<init>(TracTracRaceTrackerImpl.java:267)
com.sap.sailing.domain.tractracadapter.impl.DomainFactoryImpl.createRaceTracker(DomainFactoryImpl.java:710)
com.sap.sailing.domain.tractracadapter.impl.RaceTrackingConnectivityParametersImpl.createRaceTracker(RaceTrackingConnectivityParametersImpl.java:120)
com.sap.sailing.server.impl.RacingEventServiceImpl.addRace(RacingEventServiceImpl.java:1571)

The TrackedRegattaImpl.removeRace method here is the problem:
    `@Override`
    public void removeTrackedRace(TrackedRace trackedRace) {
        LockUtil.lockForWrite(trackedRacesLock);
        try {
            trackedRaces.remove(trackedRace.getRace());
            for (RaceListener listener : raceListeners.keySet()) {
                listener.raceRemoved(trackedRace);
            }
        } finally {
            LockUtil.unlockAfterWrite(trackedRacesLock);
        }
    }
as it acquires the write lock for trackedRacesLock and then issues listener.raceRemoved which runs into the synchronized block in AbstractTrackedRegattaAndRaceObserver.

A typical deadlock scenario. The trackedRacesLock and the AbstractTrackedRegattaAndRaceObserver monitor would have to be acquired always in the same order but aren't.

[Bug 4274](https://github.com/eclipse-sailing-analytics/sailing-analytics/issues/4274) which was trying to solve one deadlock issue has probably introduced this deadlock here. It seems that for [bug 4274](https://github.com/eclipse-sailing-analytics/sailing-analytics/issues/4274) it seemed sufficient to move a get() call on a SmartFutureCache into a separate new thread so the monitor on the RegattaListener could be released.

Looking closer at TrackedRegattaImpl shows that there is an asymmetry between addTrackedRace and removeTrackedRace. While addTrackedRace called the listeners outside the trackedRacesLock, removeTrackedRace notified the listeners while still holding the trackedRacesLock.

In commit c0d595a9c357c3fc22fcf8d4a77cdfb42b7221d3 I've made things symmetric by moving the remove listener notification outside of the lock-holding block. I'm now trying to add a failing test case.

**Comment by eclipse-sailing-analytics-bot** [comment]
**@axeluhl** commented:

bf7ce8379c6305c3617b73c2e14762fd895ddb46 on bug4414_failing_test has a failing test before the fix. Merged into [bug4414](https://github.com/eclipse-sailing-analytics/sailing-analytics/tree/bug4414) with commit 20b8c94a1f1e21dee90855598529b02cf41cc35f, and now the test is green. Letting Hudson have a look. https://hudson.sapsailing.com/job/[bug4414](https://github.com/eclipse-sailing-analytics/sailing-analytics/tree/bug4414)/
